### DISCUSSION TITLE
Installation bei 7490 nicht möglich

### DISCUSSION THREAD

**Discussion by Hoerli1337** [Freetz-NG/freetz-ng#695] [Allgemein] [answered]
Hi!
Ich habe versucht eine 7490 mit freetz-ng zu versorgen.
Dazu habe ich eine VM mit Debian 11 aufgesetzt und alle notwendigen Abhängigkeiten installiert.
Das Erstellen des Images war soweit auch erfolgreich.
Beim Aufspielen über das Webinterface, lässt mich die Box das nicht tun.
![FritzBox7490-Update-Nicht-Moeglich](https://user-images.githubusercontent.com/13108516/211205055-58a3c99c-670b-4f26-a076-0e0870f7ff78.png)
- FritzBox 7490 (DE-Version)
- Firmware 113.07.29
- Im Mesh aufgenommen
- Habe die Box vor ca. 6 Monaten mal zurückgesetzt, seitdem läuft sie durchgehend
- Konfiguration über menuconfig weitestgehend erst mal auf Standard belassen
Hier ist die Config:
[config.txt](https://github.com/Freetz-NG/freetz-ng/files/10368782/config.txt)

Muss ich noch etwas bestimmtes Einstellen, damit Freetz-NG auf der Box läuft oder hab ich gerade nur extremes Pech?
Muss ich die Firmware mit dem Flash-Tool installieren?

**Comment by pfichtner** [comment]
Über AVM's Weboberfläche können nur von AVM freigegebene/durch AVM signierte Images eingespielt werden. Du musst die Firmware über den Bootloader einspielen (flashen).

**Comment by fda77** [comment]
Da es kein Issue ist...

**Comment by Hoerli1337** [comment]
Ooooh okay dann habe ich die Beginner-Anleitung auf der Webseite leider missverstanden.
Dann versuche ich das mal mit dem Flashen.

Gibt es dazu eine passende Anleitung?

**Reply by fda77** [reply]
Das ist seit ein paar Jahren die Top-Frage

**Reply by pfichtner** [reply]
Bei meiner Suche für die [Antwort](https://github.com/Freetz-NG/freetz-ng/discussions/678#discussioncomment-4442109) auf die Frage die in die gleiche Richtung ging, bin ich auch nicht wirklich fündig geworden. 
Wenn es was gibt, ist es nicht so leicht auffindbar
Wenn es nichts gibt, könnte/sollte man das nachholen. 

Und mit "man" bist nicht Du (@fda77) gemeint sondern "jemand", und das es nicht leicht auffindbar oder nicht da ist keine Kritik sondern meine subjektive Darstellung, dass ich mir bei der Suche nach einer Bescheibung auch schwer tat.

**Reply by Hoerli1337** [reply]
Da bin ich doch froh, dass ich nicht blind bin, weil ich nichts gefunden habe :D

**Comment by fda77** [comment]
Meine Worte... irgend jemand sollte mal eine .md-Datei schreiben was es dazu wissenswertes gibt.
 - Warum
 - Links mit Anleitungen
 - von Windows/Linus aus
 Die könnte man dann unterhalb https://github.com/Freetz-NG/freetz-ng/tree/master/docs/wiki/ packen oder auch auf der Hauptseite verlinken. Und immer wenn jemand fragt einfach den Link posten

**Reply by fda77** [reply]
Das gleiche auch für die "fehlenden Kernelmodule"

**Reply by pfichtner** [reply]
Wie wäre es das in zwei Issues mit dieser Beschrebung zu packen und als "good first issue" zu tagen?

**Reply by fda77** [reply]
Ich halte nichts von offenen Issues. Entweder jemand macht was oder halt nicht... 
Bei freetz.org waren am Ende ca 230 geöffnete Tickets. Das hat nur erreicht dass man nichts mehr fand

**Reply by Hoerli1337** [reply]
Ein Hinweis das es ab Version XXXX nicht mehr geht und es auch keine passenden Scripte dafür gibt, wäre auch gut.
Ich habe gesehen, dass die neueren Modelle (xxx AX) auch langsam unterstützt werden, nur wie soll man da was drauf bekommen, wenn nichts dafür da ist.

Bin aktuell dabei gewesen, mir eine Mini-Anleitung zu schreiben, bleib aber beim Flashen stecken :(

**Reply by pfichtner** [reply]
> Ich halte nichts von offenen Issues. Entweder jemand macht was oder halt nicht... Bei freetz.org waren am Ende ca 230 geöffnete Tickets. Das hat nur erreicht dass man nichts mehr fand

Verständlich und nachvollziehbar. Jetzt verstehe ich, warum manche Projekte jeweils eigene Projekte für "issues" und "feature-requests" in ihrer Organisation haben und im eigentlich Projekt die Issues "gesperrt" haben.

**Reply by Hoerli1337** [reply]
Keine Ahnung ob man mit dem Script-Tool sich was basteln könnte:
https://fritz-tools.readthedocs.io/de/latest/

**Reply by pfichtner** [reply]
> Keine Ahnung ob man mit dem Script-Tool sich was basteln könnte: https://fritz-tools.readthedocs.io/de/latest/

Vor-/Nachbereitung des Flashvorgangs könnte man übernehmen

**Reply by fda77** [reply]
man könnte auch einfach einen Link dorthin setzen. Damit kann man aber nur die 1. der 5 verschiedenen arten flashen, also bis 7390. Nutzlos

**Reply by Hoerli1337** [reply]
Hmmmm :(
Eine 7390 wäre auch zufällig noch hier.

**Comment by fda77** [comment]
AVM hat ein tolles Feature eingebaut. Es wird angezeigt dass es kein Internet gibt, auch wenn man es gar nich aktiviert hat
![image](https://user-images.githubusercontent.com/471850/211571092-985f3d6a-3223-49a5-ac9e-ca1779a2a4eb.png)

Als Fehlerbehebung soll man das Kabel (andersrum?) reinstecken
![image](https://user-images.githubusercontent.com/471850/211572143-a5504abd-cffd-4f9b-9b76-9f78d1afb036.png)


Die Telekom bezeichnet diese Scheisse übrigens als 100 mbit. Die Labor hat leider keine Wunder bewirkt
Ist das Komma nach "USB" ein Fehler von Freetz oder "ab Werk"?

**Reply by Conan179** [reply]
"auch wenn man es gar nich aktiviert hat"? Wie definierst du das? Ich sehe, das sie kontackt mit der Vermittlungstelle hat und aus gehandelt hat, wie schnell gesufrt werden darf, nur die login daten scheinen nicht zu stimmen.

**Reply by fda77** [reply]
Die ist nur Modem, als "kein Logindaten" oder so ähnlich ausgewählt. PPPOE macht ein richtiger Router :)
Diese Warnbox kommt übrigens nicht sofort sondern erst nach einer längeren Zeit

**Reply by Conan179** [reply]
Die Meldung kenne ich, ich hab hier Kabel Fritten, wen da für mehr als 59 kein kabel signal an liegt kommt auch die Meldung ;) Dafür brauche ich keine Log in Daten :D. ( ich warte nur drauf, das avm die 7.50 für meine 6690 veröffentlicht, dann wird sie gefreetzt)

**Comment by fda77** [comment]
@Hoerli1337: Schau mal ob dir das hilft: https://freetz-ng.github.io/freetz-ng/INSTALL

**Reply by Hoerli1337** [reply]
Das kann ich mal übers Wochenende testen.
`Gen 3: ram-boot / NAND / inmemory `    sollte passen, mal sehen ob das mit der aktuellsten Firmware noch geht.

**Comment by Hoerli1337** [comment, marked as answer]
Fürs Protokoll / Wiki hier die passenden Schritte für eine FRITZ!Box 7490
1. Box stromlos machen
2. Box direkt mit dem PC verbinden (RJ45)
3. Auf dem PC das push_firmware-Script ausführen und mit Y bestätigen
`# tools/push_firmware -mr`
4. Box mit Strom versorgen
5. Warten
6. Erfolg
![FritzBox 7490 Push Firmware](https://user-images.githubusercontent.com/13108516/212347990-69a3a233-9074-4032-a2a5-5125d03a618e.png)
![FritzBox 7490 Push Firmware Webinterface](https://user-images.githubusercontent.com/13108516/212348002-7f5d1f5e-c69a-42f8-89bb-de6ab91988f8.png)
![FritzBox 7490 Push Firmware Webinterface 2](https://user-images.githubusercontent.com/13108516/212348032-251dd43f-0425-4247-9e08-f5591a2cab93.png)

Webinterface:
Benutzer: admin
Passwort: freetz

SSH:
Benutzer: root
Passwort: freetz

Getestet mit:
- Firmware: 113.07.29 rev92201 
- Box war bereits im Netzwerk und hat nicht die IP 192.168.178.1
- Box war im Mesh aufgenommen und war es im Anschluss immer noch

**Reply by pfichtner** [reply]
>     2. Box direkt mit dem PC verbinden (RJ45)

Wie @fda77 schrieb, das "direkt" ist nicht zwindend notwendig (und mit media sensing o.ä. u.U. sogar nachteilig). Meines Wissens nach muss es aber zwingend die "LAN1"-Buchse der FritzBox sein.

**Reply by fda77** [reply]
Nö, ist egal. Ich glaub AVM hat das nur hingeschrieben um der Frage "Und in welchen Port???" zuvorzukommen...

**Comment by fda77** [comment]
Prima, fehlt auf der INSTALL Seite noch irgend was grundlegendes? Oder hast du noch gute Links für dorthin?

> tools/push_firmware -mr

Das "-mr" braucht man nicht unbedingt: https://freetz-ng.github.io/freetz-ng/INSTALL.html#push_firmware
> Wenn man nicht weiss weshalb man einen Parameter angegeben halt sollte man diesen weglassen! Es müsste alles automatisch erkannt werden.

> Box direkt mit dem PC verbinden (RJ45)

Das hab ich noch nie gemacht, mit meinem Switch ist das kein Problem. Powerline und WLan sind nicht so optimal

**Reply by Hoerli1337** [reply]
Sonst ist mir nichts mehr sonderbares aufgefallen.
Habe alle weiteren Infos wie z.B. zum erstellen des Images auf der Seite gefunden.

**Comment by fda77** [comment]
Danke für die Rückmeldung. Du könntest schonmal bei [AVM](mailto:fritzbox_info@avm.de) anfragen:

```
  generating modules.dep
depmod: WARNING: build/modified/filesystem/lib/modules/3.10.107/extra/wireguard.ko needs unknown symbol avm_pa_reset_skb
depmod: WARNING: build/modified/filesystem/lib/modules/3.10.107/kernel/extra/kdsld/kdsldmod.ko needs unknown symbol avm_pa_reset_skb
depmod: WARNING: build/modified/filesystem/lib/modules/3.10.107/kernel/extra/kdsld/kdsldmod.ko needs unknown symbol avm_pa_pid_set_bridging
depmod: WARNING: build/modified/filesystem/lib/modules/3.10.107/kernel/extra/kdsld/kdsldmod.ko needs unknown symbol _avm_pa_dev_snoop_transmit
depmod: WARNING: build/modified/filesystem/lib/modules/3.10.107/kernel/extra/speedtest/kspeedtest.ko needs unknown symbol avm_pa_add_drop_session
WARNING: Unresolved symbols detected, not all AVM-features may work.
No current sources by AVM? Error in kernel's .config? Ask fritzbox_info@avm.de for sources!
```
